internal/fleet/agent_policy_resource.go

```diff
@@ -0,0 +1,284 @@
+package fleet
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/fleet"
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/fleet/fleetapi"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	monitorLogs    = "logs"
+	monitorMetrics = "metrics"
+)
+
+func ResourceAgentPolicy() *schema.Resource {
+	agentPolicySchema := map[string]*schema.Schema{
+		"id": {
+			Description: "Internal identifier of the resource.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"policy_id": {
+			Description: "Unique identifier of the agent policy.",
+			Type:        schema.TypeString,
+			Computed:    true,
```

Comment:
```suggestion
			Computed:    true,
			Optional:       true, 
			ForceNew:     true,
```

Should users be able to set this as well? Reading the create code it feels like they should. Also, if they can then changing the ID should force a new resource.